arcgis-ios-sdk-samples/Maps/Read a geopackage/ReadGeopackageViewController.swift

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+
+class ReadGeopackageViewController: UIViewController {
+    
+    @IBOutlet weak var mapView:AGSMapView!
+    
+    private var geoPackage:AGSGeoPackage?
+    private var allLayers:[AGSLayer] = []
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Instantiate and display a map using a basemap, location, and zoom level.
+        mapView.map = AGSMap(basemapType: .streets, latitude: 39.7294, longitude: -104.8319, levelOfDetail: 11)
+        
+        // Create a geopackage from a named bundle resource.
+        geoPackage = AGSGeoPackage(name: "AuroraCO")
+        
+        // Load the geopackage.
+        geoPackage?.load { [weak self] error in
+            guard error == nil else {
+                SVProgressHUD.showError(withStatus: "Error opening Geopackage: \(error!.localizedDescription)", maskType: .gradient)
+                return
+            }
+
+            // Create feature layers for each feature table in the geopackage.
+            let featureLayers = self?.geoPackage?.geoPackageFeatureTables.map({ featureTable -> AGSLayer in
+                return AGSFeatureLayer(featureTable: featureTable)
+            }) ?? []
+            
+            // Create raster layers for each raster in the geopackage.
+            let rasterLayers = self?.geoPackage?.geoPackageRasters.map({ raster -> AGSLayer in
+                return AGSRasterLayer(raster: raster)
+            }) ?? []
+
+            // Keep an array of all the feature layers and raster layers in this geopackage.
+            self?.allLayers.append(contentsOf: featureLayers)
```

Comment:
can you add the raster layers first and then feature layers? that way in the UI the order I add layers going top to bottom will be more preferable, raster layers will get added first and feature layers on top. This will allow me to see all the layers being added. Right now, the raster layers get added last and obscure feature layers underneath

Reply:
Good idea. Rasters are now listed first before features.